pkg/kubelet/pod_workers.go

```diff
@@ -1245,15 +1247,22 @@ func (p *podWorkers) podWorkerLoop(podUID types.UID, podUpdates <-chan struct{})
 				// when we receive a running pod, we don't need status at all because we are
 				// guaranteed to be terminating and we skip updates to the pod
 			default:
-				// wait until we see the next refresh from the PLEG via the cache (max 2s)
-				// TODO: this adds ~1s of latency on all transitions from sync to terminating
-				//  to terminated, and on all termination retries (including evictions). We should
-				//  improve latency by making the pleg continuous and by allowing pod status
-				//  changes to be refreshed when key events happen (killPod, sync->terminating).
-				//  Improving this latency also reduces the possibility that a terminated
-				//  container's status is garbage collected before we have a chance to update the
-				//  API server (thus losing the exit code).
-				status, err = p.podCache.GetNewerThan(update.Options.Pod.UID, lastSyncTime)
+				if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) {
+					status, err = p.podCache.Get(update.Options.Pod.UID)
+					if err == nil && status.TimeStamp.Before(lastSyncTime) {
```

Comment:
There is something wrong here. This does the same thing as just calling `GetNewerThan()` from the beginning.

Reply:
Would it make sense to implement eventedpleg.UpdateCache() so that it updates the global timestamp with the current timestamp on successful update, and call that instead of the generic? There does not seem to be a way to safely change the check here